src/particle.cpp

```diff
@@ -272,6 +273,10 @@ void Particle::event_cross_surface()
   } else {
     // Particle crosses surface
     cross_surface();
+    if (settings::weight_windows_on &&
+        settings::weight_window_checkpoint_surface) {
+      apply_weight_windows(*this);
+    }
```

Comment:
To clean this up further, let's push the check for `settings::weight_windows_on` into `apply_weight_windows` so the first line of that function is:

```cpp
if (!settings::weight_windows_on) return;
```

Reply:
addressed in the latest commit